### PR TITLE
Separate building library and extension

### PR DESCRIPTION
**Pull request by hickford**
Hi Mike. I recently learnt how to separate building library and extension in setup.py . I'm not convinced it's any better or worse, but thought I'd share. What do you think?

### CODE REVIEW

**Comment by superbobry** [comment]
Hello, I like the idea. Does this put any restrictions on the `setuptools` version?

File: setup.py [hunk 1]
```diff
@@ -64,12 +64,18 @@ def run_tests(self):
     license=LICENSE,
     url='https://github.com/kmike/datrie',
     classifiers=CLASSIFIERS,
+    libraries=[
+        ('libdatrie', dict(
+        sources=LIBDATRIE_FILES,
```
Comment (superbobry) [inline setup.py]: Can you please PEP-8 this?

**Comment by hickford** [comment]
Cool. Not that I know. The boon is "building libdatrie library" and "building datrie extension" show as separate steps in the install.

I learnt it from http://stackoverflow.com/q/16854066/284795

File: setup.py (hunk 1 above)
Comment (hickford) [inline setup.py]: Sure. Does that mean single or double quotes?

Comment (superbobry) [inline setup.py]: Actually I meant indentation in the `dict` call.

**Comment by hickford** [comment]
Better?

**Comment by superbobry** [comment]
Looks good, thanks!

**Pull request merged**